source/Calamari.Tests/Fixtures/Integration/Packages/NuGetPackageDownloaderFixture.cs

```diff
@@ -59,5 +63,76 @@ public int AttemptsTheRightNumberOfTimesOnError(int maxDownloadAttempts)
 
             return calledCount;
         }
+
+#if USE_NUGET_V2_LIBS
+        [Test]
+        public void TimesOutIfAValidTimeoutIsDefinedInVariables()
+        {
+            RunNugetV3TimeoutTest("00:00:01", TimeSpan.FromSeconds(10), TimeSpan.FromSeconds(1));
+        }
+
+        [Test]
+        public void IgnoresTheTimeoutIfAnInvalidTimeoutIsDefinedInVariables()
+        {
+            RunNugetV3TimeoutTest("this is not a valid timespan", TimeSpan.FromSeconds(2), TimeSpan.FromSeconds(2));
+        }
+
+        [Test]
+        public void DoesNotTimeOutIfNoTimeoutIsDefinedInVariables()
+        {
+            RunNugetV3TimeoutTest(null, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(1));
+        }
+
+        [Test]
+        public void DoesNotTimeOutIfTheServerRespondsBeforeTheTimeout()
+        {
+            RunNugetV3TimeoutTest("00:01:00", TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(1));
+        }
+
+        void RunNugetV3TimeoutTest(string timeoutInVariables, TimeSpan serverResponseTime, TimeSpan estimatedTimeout)
+        {
+            using (var server = new TestHttpServer(9001, serverResponseTime))
```

Comment:
Do these tests get run in parallel? Will this clash?

Reply:
They don't get run in parallel. But to protect us in future if we ever do try to turn on parallel tests, I've added the  `[NonParallelizable]` attribute to them all.